src/client/opamArg.mli

```diff
@@ -25,15 +25,18 @@ val cli2_0: OpamCLIVersion.t
 val cli2_1: OpamCLIVersion.t
 val cli2_2: OpamCLIVersion.t
 
-(* [cli_from since] validity flag since [since], and no removal version *)
-val cli_from: OpamCLIVersion.t -> validity
+(* [cli_from ?experimental since] validity flag since [since], and no removal
+   version. If experimental is given, it is marked as is (warning and doc
+   update). *)
+val cli_from: ?experimental:bool -> OpamCLIVersion.t -> validity
 
-(* [cli_between since until ?replaced] a validity flags introduced in
+(* [cli_between ?experimental since until ?replaced] a validity flags introduced in
    [since], removed in [until], [replaced] is the replacement helper
-   message *)
+   message. If experimental is given, it is marked as is (warning and doc
+   update). *)
 val cli_between:
-  OpamCLIVersion.t -> ?default:bool -> ?replaced:string -> OpamCLIVersion.t ->
-  validity
+  ?experimental:bool -> OpamCLIVersion.t -> ?default:bool -> ?replaced:string ->
```

Comment:
Do we want experimental on removed flags ?

Reply:
I don't think we need that, no - if we remove an experimental flag, then it's highly like that calling that flag (even with `--cli=`) is going to result in an error message, and it's there that we'd state that "The experimental flag --magically-solve-everything isn't supported by this release of opam".